### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by MikaelSmith**
Update all dependencies to latest versions. In particular, brings context support to Kubernetes API calls.

Also switches from fuse to syscall error constants for bazil/fuse@e2f0bde

Signed-off-by: Michael Smith <michael.smith@puppet.com>

### CODE REVIEW

**Comment by MikaelSmith** [comment]
Easiest to wait until after #762 to merge.

File: fuse/dir.go [hunk 1]
```diff
@@ -38,7 +39,7 @@ func (d *dir) children(ctx context.Context) (*plugin.EntryMap, error) {
 		return plugin.ListWithAnalytics(ctx, updatedEntry.(plugin.Parent))
 	}
 
-	return nil, fuse.ENOENT
+	return nil, syscall.ENOENT
```
Comment (ekinanp) [inline fuse/dir.go]: Why this change?

**Review by ekinanp — APPROVED** [review]

File: fuse/dir.go (hunk 1 above)
Comment (MikaelSmith) [reply fuse/dir.go]: https://github.com/bazil/fuse/commit/e2f0bde20aef2c6f9a3da6d1d2d3cf2abf4ebe2b

Comment (MikaelSmith) [reply fuse/dir.go]: Actually that deprecated all `fuse.E*`. I'll update those and mention it in the commit message.

**Pull request merged**